### DISCUSSION TITLE
Wanted Features Thread  No. 1

### DISCUSSION THREAD

**Discussion by andynameistaken** [cmacrae/spacebar#75] [Ideas]
I added  number because if list becomes too long, the discussion  can be closed and we can start another so it will look nice and clean.

Wanted features:
* Support for PowerLine fonts 
![Screenshot 2021-04-25 at 23 56 22](https://user-images.githubusercontent.com/15692002/116011004-d6eeb880-a622-11eb-8bad-8d34dbc8e2dc.png)

* Plugin folder - allowing to add plugins by other users would be useful
* Button for donations in `README.md` - it would be nice to support this project so you could spend more time on it and be able to tell someone at the party (or someone you are fond of) - "I get money from OpenSource project" and be cool 😎
* Adding `echo`  about donation in `brew` install script
* Install script could check if user has required fonts for displaying default config and if not it could ask about installing them
* Mouse support for plugins and workspaces - because why not
* Adding more visual options - for example Kawase blur and colour gradients
* Adding support for `iStats Menus`

**Comment by cmacrae** [comment]
Thanks for starting this discussion @andynameistaken 🙌   

#### PowerLine fonts
Have you tried this? If it's just a font, it should work :) But perhaps I'm misunderstand the intended use?

#### Plugin folder
Great idea! I think maybe once we've built up a few more we can add this 👍 

#### Button for donations in `README.md`  
>it would be nice to support this project so you could spend more time on it and be able to tell someone at the party (or someone you are fond of) - "I get money from OpenSource project" and be cool 😎
😆 

Thanks so much for this suggestion 🙇 It really means a lot to me that there are people out there who'd be interested in this.  
I set this up earlier today, so now there's 'Sponsor' butons in the repo UI and a badge in the README. It's really hard trying to word the profile and tiers descriptions so you don't come across like you just want money!

#### Message in brew install about sponsorship
Whilst I *really* appreciate the incentive 🤗 I'd prefer not to do this, personally.  
My software will always be free, and while I know I could word it such that it didn't imply payment was necessary, I'd just perfer not to blur that line.

#### Font checking
I'd like to leave this sort of thing out of the primary offerings of the repo (and package repo).  
From a standpoint of purity, only the software produced and directly associated material should be provided.  
I could, however, add some logging to emit a friendly error when a font isn't found :)

#### Mouse support
This is something I'm looking into in the ongoing rewrite! 🐁 

#### Adding more visual options 
This could be really cool! I'll have to see what capabilities are granted in CoreGraphics, as that's what everything's built off.  
I'll certainly look into this for the rewrite.

#### Adding support for iStats Menus
Hmm, I'm not so sure this'd be possible. It's an assumption (but I pretty solid one, I feel) that this would - at its core - rely on drawing to the native macOS menubar, which is a completely separate entity to spacebar.

 From a brief look at this tool, it seems like it does some quite nice, fancy things. This project is supposed to be a minimal implementation of a status bar - akin to those used in Linux/BSD tiling window environments - not only do I want to keep it this way, but I certainly don't currently have the knowledge of the macOS frameworks necessary to implement something like this 😅
